crates/topos/tests/config.rs

```diff
@@ -123,3 +123,90 @@ async fn test_handle_command_init_with_custom_name() -> Result<(), Box<dyn std::
 
     Ok(())
 }
+
+/// Test node init arguments precedence
+/// •  CLI flag should overwrite ENV variable
+#[tokio::test]
+async fn test_command_init_precedence() -> Result<(), Box<dyn std::error::Error>> {
+    // Test node init with env variables
+    let node_init_home_env = "/tmp/topos/test_command_init_precedence_env";
+    let node_edge_path_env = polygon_edge_path(node_init_home_env).await;
+    let node_init_name_env = "TEST_NODE_ENV";
+    let node_init_role_env = "full-node";
+    let node_init_subnet_env = "topos-env";
+
+    let mut cmd = Command::cargo_bin("topos")?;
+    cmd.arg("node")
+        .env("TOPOS_POLYGON_EDGE_BIN_PATH", &node_edge_path_env)
+        .env("TOPOS_HOME", node_init_home_env)
+        .env("TOPOS_NODE_NAME", node_init_name_env)
+        .env("TOPOS_NODE_ROLE", node_init_role_env)
+        .env("TOPOS_NODE_SUBNET", node_init_subnet_env)
+        .arg("init");
+
+    let output = cmd.assert().success();
+    let result: &str = std::str::from_utf8(&output.get_output().stdout)?;
+
+    // Test node init with cli flags
+    assert!(result.contains("Created node config file"));
+    let home = PathBuf::from(node_init_home_env);
+    // Verification: check that the config file was created
+    let config_path = home
+        .join("node")
+        .join(node_init_name_env)
+        .join("config.toml");
+    assert!(config_path.exists());
+    // Check if config file params are according to env params
+    let config_contents = std::fs::read_to_string(&config_path).unwrap();
+    assert!(config_contents.contains("name = \"TEST_NODE_ENV\""));
+    assert!(config_contents.contains("role = \"fullnode\""));
+    assert!(config_contents.contains("subnet = \"topos-env\""));
+    std::fs::remove_dir_all(node_init_home_env)?;
```

Comment:
We should use [`tempfile`](https://crates.io/crates/tempfile) instead of creating and cleaning up temp files. E.g. if one of the `assert`s above here fail, would the file be cleaned up?